### PR TITLE
Redirected user if 403 error received

### PR DESCRIPTION
**Pull request by sudeeptarlekar**
Signed-off-by: sudeeptarlekar <sudeeptarlekar@gmail.com>

## Notion card
https://www.notion.so/saeloun/Employees-shouldn-t-be-able-to-access-the-project-detail-page-ced94d36a30644edb7a0108aa24ab845

## Summary
Redirect user back to index pages if error returned by Apis.

## Type of change
- [ x ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Tested on local setup, with employee record in database

### Checklist:

- [ x ] I have manually tested all workflows
- [ x ] I have performed a self-review of my own code

### CODE REVIEW

**Review by onkar-saeloun — APPROVED** [review]
LGTM

**Comment by apoorv1316** [comment]
Screenshot 2022-09-13 at 11 33 45 AM
@sudeeptarlekar If a user enters a URL like project/:id then it navigates to the project but also throws a toast error here. I don't think he should be shown a toast message. Just the redirect would work. The same thing for clients also. @supriya3105 What do you think about this?

**Comment by sudeeptarlekar** [comment]
How would user know, why he got redirected?

**Comment by apoorv1316** [comment]
> How would user know, why he/she got redirected?

If we wanna do it this way then the error toast message should be user-friendly. Also if an employee visits invoices or payments he/she is just redirected. So whatever we decide can we make the behavior consistent for all the routes? cc @supriya3105

**Comment by supriya3105** [comment]
> > How would user know, why he/she got redirected?
> 
> If we wanna do it this way then the error toast message should be user-friendly. Also if an employee visits invoices or payments he/she is just redirected. So whatever we decide can we make the behavior consistent for all the routes? cc @supriya3105

We should show the toast error message. The message should let the user know that they cannot access the page as they do not have the permission for it. I will add a design for this with correct error message. Also, the same should be applied across all pages. 
@sudeeptarlekar @apoorv1316

**Comment by sudeeptarlekar** [comment]
@supriya3105 are you creating a new ticket for message, or should I add changes for same here?

File: app/javascript/src/components/Clients/Details/index.tsx [hunk 1]
```diff
@@ -78,6 +79,10 @@ const ClientList = ({ isAdminUser }) => {
         setProjectDetails(sanitized.projectDetails);
         setTotalMinutes(sanitized.totalMinutes);
         setOverDueOutstandingAmt(sanitized.overdueOutstandingAmount);
+      })
+      .catch((e) => {
+        console.log(e); // eslint-disable-line
```
Comment (akhilgkrishnan) [inline app/javascript/src/components/Clients/Details/index.tsx:84]: You can use logger instead of console log.

Comment (sudeeptarlekar) [reply app/javascript/src/components/Clients/Details/index.tsx]: fixed

**Review by akhilgkrishnan — APPROVED** [review]
LGTM

**Review by apoorv1316 — APPROVED** [review]

**Comment by supriya3105** [comment]
> @supriya3105 are you creating a new ticket for message, or should I add changes for same here?

@apoorv1316 @sudeeptarlekar 

In case the user tries to access the project or client details page using the URL, the redirect the user to project list or client list page respectively. Display a toast message saying that 

“You are not authorized to access the page. Please reach out to your administrator for help.”

**Comment by sudeeptarlekar** [comment]
project_client

Fixed

**Comment by sudeeptarlekar** [comment]
Are we still missing something here? Is it good to merge.

cc: @supriya3105

**Comment by supriya3105** [comment]
@sudeeptarlekar Apoorv is testing the multiple workspace scenario today. accordingly to the test results, we will merge this PR.

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**